File: src/ceuCoin/Transaction.java
package ceuCoin;

import java.time.LocalDateTime;

public class Transaction {
    public String sender;
    public String recipient;
    public double amount;
    public LocalDateTime datetime;

    public Transaction(String sender, String recipient, double amount) {
        this.sender = sender;
        this.recipient = recipient;
        this.amount = amount;
        this.datetime = LocalDateTime.now();
    }

    public Transaction(String sender, String recipient, double amount, LocalDateTime datetime) {
        this.sender = sender;
        this.recipient = recipient;
        this.amount = amount;
        this.datetime = datetime;
    }

    public String getSender() { return sender; }
    public String getRecipient() { return recipient; }
    public double getAmount() { return amount; }
    public LocalDateTime getDatetime() { return datetime; }

    @Override
    public String toString() {
        return "ceuCoin.Transaction{" +
                "sender='" + sender + '\'' +
                ", recipient='" + recipient + '\'' +
                ", amount=" + amount +
                ", date=" + datetime +
                '}';
    }
}

File: ceu/Transaction.py
from datetime import datetime

class Transaction:
    def __init__(self, sender, recipient, amount):
        self.sender = sender
        self.recipient = recipient
        self.amount = amount
        self.datetime = datetime.now()

    def __str__(self):
        return f"ceuCoin.Transaction{{sender='{self.sender}', recipient='{self.recipient}', amount={self.amount}, datetime={self.datetime}}}"

    def get_sender(self):
        return self.sender

    def get_recipient(self):
        return self.recipient

    def get_amount(self):
        return self.amount

    def get_datetime(self):
        return self.datetime
